### ISSUE TITLE
Undefined tmpImg Bug

### ISSUE THREAD

**Issue by vladtarrow** [darsain/imagesloaded#1]
1. Dom with broken images was subscribed to method ImagesLoaded.fail.
2. Dom was changed: some images were removed from Dom.
3. After several seconds ImagesLoaded.fail performing started.
4. Bug: _element_ is **undefined**
```
  function listener(element, eventName, handler, remove) {
        var events = eventName.split(' ');
        for (var i = 0, l = events.length; i < l; i++) {
            if (element.addEventListener) {
                element[remove ? 'removeEventListener' : 'addEventListener'](events[i], handler, false);
            } else if (element.attachEvent) {
                element[remove ? 'detachEvent' : 'attachEvent']('on' + events[i], handler);
            }
        }
    }
```

This function was called by imgLoadedHandler(event) method, which has `this[ILID].tmpImg` as undefined.
5. Proposed HotFix : 
```
 function imgLoadedHandler(event) {
            /*jshint validthis:true */
            event = event || w.event;
            // Unbind loaded handler from temporary image
           unbind(this[ILID].tmpImg || {}, EVENTS, imgLoadedHandler);
            // Leave the temporary image for garbage collection
            this[ILID].tmpImg = null;
            // Don't proceed if image is already loaded
            if (inArray(this, self.loaded) === -1) {
                imgLoaded(this, event.type !== 'load');
            }
        }
```
line `unbind(this[ILID].tmpImg || {}, EVENTS, imgLoadedHandler);`
If `this[ILID].tmpImg` is `undefined `to send empty Object.

**Comment by darsain** [comment]
Eh, I don't think I want to maintain this anymore. I added you as a collaborator. Feel free to apply any fixes :)

**Comment by vladtarrow** [comment]
fixed 08fc5418b1296202fd22245a6d6241981239caee
